### PR TITLE
Feature: add cap amount for naked margin collateral

### PR DESCRIPTION
**Pull request by haythem96**
# Feature: add cap amount for naked margin collateral

## High Level Description

This PR add a cap amount for the collateral deposited in naked margin vault.
The cap amount is set per collateral asset, and should be scaled by the collateral asset decimals.

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [x] Is your code up to date with the spec? 
- [ ] Have you added your tests to the testing doc?

### CODE REVIEW

**Review by CruzMolina — APPROVED** [review]
✅

File: contracts/core/MarginCalculator.sol [hunk 1]
```diff
@@ -233,12 +252,21 @@ contract MarginCalculator is Ownable {
...
+     * @return dust amount
      */
     function getCollateralDust(address _collateral) external view returns (uint256) {
         return dust[_collateral];
     }
 
+    /**
+     * @notice get cap amount for collateral asset
+     * @param _collateral collateral asset address
+     * @return dust amount
```
Comment (antoncoding) [inline contracts/core/MarginCalculator.sol:264]:
```suggestion
     * @return cap amount
```

File: contracts/core/MarginCalculator.sol [hunk 2]
```diff
@@ -640,18 +668,29 @@ contract MarginCalculator is Ownable {
...
             if (_vaultDetails.vaultType == 1) {
                 // this is a naked margin vault
-                // fetch dust amount for otoken collateral asset as FixedPointInt, assuming dust is already scaled to 1e27
+                // fetch dust amount for otoken collateral asset as FixedPointInt, assuming dust is already scaled by collateral decimals
                 FPI.FixedPointInt memory dustAmount = FPI.fromScaledUint(
                     dust[_vaultDetails.shortCollateralAsset],
                     _vaultDetails.collateralDecimals
                 );
 
+                // fetch collateral cap amount for otoken collateral asset as FixedPointInt, assuming dust is already scaled by collateral decimals
```
Comment (antoncoding) [inline contracts/core/MarginCalculator.sol:677]:
```suggestion
                // fetch collateral cap amount for otoken collateral asset as FixedPointInt, assuming cap is already scaled by collateral decimals
```

**Review by antoncoding — APPROVED** [review]

File: contracts/core/MarginCalculator.sol [hunk 3]
```diff
@@ -111,6 +116,20 @@ contract MarginCalculator is Ownable {
         emit CollateralDustUpdated(_collateral, _dust);
     }
 
+    /**
+     * @notice set cap amount for collateral asset used in naked margin
+     * @dev can only be called by owner
+     * @param _collateral collateral asset address
+     * @param _cap dust amount, should be scaled by collateral asset decimals
```
Comment (CruzMolina) [inline contracts/core/MarginCalculator.sol:123]:
```suggestion
     * @param _cap cap amount, should be scaled by collateral asset decimals
```

**Pull request merged**